packages/x-adapter/src/schemas/utils.ts

```diff
@@ -16,32 +16,30 @@ const mutableSchemasInternalMethods = ['$replace', '$override', '$extends', 'toS
  *
  * @public
  */
-export function createMutableSchema<T extends Schema>(schema: T): MutableSchema<T> {
+export function createMutableSchema<Source, Target>(
+  schema: Schema<Source, Target>
+): MutableSchema<Source, Target> {
   return {
     ...schema,
-    $replace: function <Source = any, Target = any>(
-      newSchema: Schema<Source, Target>
-    ): MutableSchema<Schema<Source, Target>> {
-      Object.keys(this).forEach(key => {
-        if (isInternalMethod(key)) {
+    $replace(newSchema) {
+      forEach(this, key => {
+        if (isInternalMethod(key as string)) {
```

Comment:
I'm willing to remove `isInternalMethod` using either a class, defining these methods as non iterable, or using a custom iterator function.